### ISSUE TITLE
WindowsError: [Error 5] Access is denied: 'C:/Program Files (x86)/microdrop\\support' (Trac #79)

### ISSUE THREAD

**Issue by cfobel** [wheeler-microfluidics/microdrop#83]
The following error was encountered when running an installed microdrop.exe on a Windows 7 computer:

$ ./microdrop.exe
Found embedded PKG: C:\Program Files (x86)\microdrop\microdrop.exe
Extracting binaries
manifestpath: C:\Program Files (x86)\microdrop\microdrop.exe.manifest
Error activating the context
C:\Program Files (x86)\microdrop\python27.dll
Manipulating evironment
PYTHONPATH=C:/Program Files (x86)/microdrop
importing modules from CArchive
extracted iu
extracted struct
extracted archive
Installing import hooks
outPYZ1.pyz
Running scripts
Traceback (most recent call last):
 File "<string>", line 31, in <module>
 File "C:\buildbot\microdrop\WinXP_Py2_7\build\build\pyi.win32\microdrop\outPYZ
1.pyz/os", line 150, in makedirs
 File "C:\buildbot\microdrop\WinXP_Py2_7\build\build\pyi.win32\microdrop\outPYZ
1.pyz/os", line 157, in makedirs
WindowsError: [Error 5] Access is denied: 'C:/Program Files (x86)/microdrop\\support'
RC: -1 from win32comgenpy
OK.

This appears to be related to this problem:

http://permalink.gmane.org/gmane.comp.python.pyinstaller/1267

As a possible temporary workaround, users could try to run the program as an administrator.

Migrated from http://microfluidics.utoronto.ca/microdrop/ticket/79
```json
{
    "status": "closed", 
    "changetime": "2014-04-17T19:39:01", 
    "description": "The following error was encountered when running an installed microdrop.exe on a Windows 7 computer:\n\n$ ./microdrop.exe\nFound embedded PKG: C:\\Program Files (x86)\\microdrop\\microdrop.exe\nExtracting binaries\nmanifestpath: C:\\Program Files (x86)\\microdrop\\microdrop.exe.manifest\nError activating the context\nC:\\Program Files (x86)\\microdrop\\python27.dll\nManipulating evironment\nPYTHONPATH=C:/Program Files (x86)/microdrop\nimporting modules from CArchive\nextracted iu\nextracted struct\nextracted archive\nInstalling import hooks\noutPYZ1.pyz\nRunning scripts\nTraceback (most recent call last):\n File \"<string>\", line 31, in <module>\n File \"C:\\buildbot\\microdrop\\WinXP_Py2_7\\build\\build\\pyi.win32\\microdrop\\outPYZ\n1.pyz/os\", line 150, in makedirs\n File \"C:\\buildbot\\microdrop\\WinXP_Py2_7\\build\\build\\pyi.win32\\microdrop\\outPYZ\n1.pyz/os\", line 157, in makedirs\nWindowsError: [Error 5] Access is denied: 'C:/Program Files (x86)/microdrop\\\\support'\nRC: -1 from win32comgenpy\nOK.\n\nThis appears to be related to this problem:\n\nhttp://permalink.gmane.org/gmane.comp.python.pyinstaller/1267\n\nAs a possible temporary workaround, users could try to run the program as an administrator.", 
    "reporter": "ryan", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1397763541728826", 
    "component": "microdrop", 
    "summary": "WindowsError: [Error 5] Access is denied: 'C:/Program Files (x86)/microdrop\\\\support'", 
    "priority": "major", 
    "keywords": "", 
    "version": "0.1", 
    "time": "2012-03-17T01:15:15", 
    "milestone": "Microdrop 1.0", 
    "owner": "ryan", 
    "type": "defect"
}
```

**Comment by cfobel** [comment]
Trac update at 20120327T17:08:35: ryan commented:

> Just want to confirm that giving the user rights to write to the "Program Files\microdrop\support" directory fixes the problem. Maybe we could install the support directory at install time? I'm not clear whether or not the files in there change or whether they are just written a single time.

**Comment by cfobel** [comment]
Trac update at 20120329T00:28:14: cfobel commented:

> ```
> commit 485b66525c8558cce18c9084cefc2bc2f9afea00
> Author: Christian Fobel <christian@fobel.net>
> Date:   Wed Mar 28 18:41:47 2012 -0400
> 
>     Install support directory with EXE
>     
>     This might fix the permissions error from ticket #83.
> ```
> 
> I cannot be reproduce this issue locally, but this commit might fix the problem.  Any chance we can try things out on the system that produced this error?

**Comment by cfobel** [comment]
Trac update at 20120329T14:26:40: 
* ryan commented:

> Confirming that this fixed the problem.


* ryan changed resolution from "" to "fixed"
* ryan changed status from "new" to "closed"

**Comment by cfobel** [comment]
Trac update at 20121105T22:06:31: 
* ryan commented:

> A related bug is occurring with the pygst_utils_windows_server.
> 
> From the traceback:
> 
> PYTHONPATH=C:/Program Files (x86)/microdrop/pygst_utils_windows_server
> importing modules from CArchive
> extracted iu
> extracted struct
> extracted archive
> Installing import hooks
> outPYZ1.pyz
> Running scripts
> Traceback (most recent call last):
>   File "<string>", line 31, in <module>
>   File "C:\Users\ryan\Documents\dev\pygst_utils\build\pyi.win32\test_video_serve
> r\outPYZ1.pyz/os", line 150, in makedirs
>   File "C:\Users\ryan\Documents\dev\pygst_utils\build\pyi.win32\test_video_serve
> r\outPYZ1.pyz/os", line 157, in makedirs
> WindowsError: [Error 5] Access is denied: 'C:/Program Files (x86)/microdrop/pygs
> t_utils_windows_server\\support'
> RC: -1 from win32comgenpy
> OK.


* ryan changed resolution from "fixed" to ""
* ryan changed status from "closed" to "reopened"

**Comment by cfobel** [comment]
Trac update at 20121119T18:53:18: 
* ryan commented:

> Fixed in commit 4f985e504dd94a08da769762bfe47f1fed21525c.


* ryan changed resolution from "" to "fixed"
* ryan changed status from "reopened" to "closed"

**Comment by cfobel** [comment]
Trac update at 20140417T19:39:01: ryan changed component from "core" to "microdrop"
